EFCore/ASP.NetCore/Blazor.WebAssembly/Readme.md

```diff
@@ -0,0 +1,1114 @@
+      } catch(AuthenticationException) {
+         return Unauthorized("User name or password is incorrect.");
+      }
+   }
+   ```
+
+4. In the same file, create a `LogoutAsync` endpoint used to log out and invalidate the cookie.
+
+   _API/Security/AuthenticationController.cs_:
+   ```cs
+   [HttpPost(nameof(LogoutAsync))]
+   public async Task<SignOutResult> LogoutAsync() {
+      await HttpContext.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
+      return new SignOutResult(CookieAuthenticationDefaults.AuthenticationScheme);
+   }
+   ```
+
+5. Create an authorized `UserInfo` endpoint used to get user information and validate the cookie from the client when needed (for example, when a user refreshes the page).
+
+   _API/Security/AuthenticationController.cs_:
+   ```cs
+   [Authorize][HttpGet(nameof(UserInfo))]
+   public ActionResult UserInfo() {
+      if (HttpContext.User.Identity?.IsAuthenticated == null || !HttpContext.User.Identity.IsAuthenticated)
+         return Unauthorized();
+      var claims = ((ClaimsIdentity)HttpContext.User.Identity).Claims.ToArray();
+      return Ok(new {
+         UserName= HttpContext.User.Identity.Name, 
+         ID = claims.First(claim => claim.Type == nameof(ApplicationUser.ID)).Value.ToString(),
+         IsActive = claims.First(claim => claim.Type == nameof(ApplicationUser.IsActive)).Value.ToString()
+      });
+   }
+   ```
+
+### Use Swagger UI to test the Web API service
+
+At this point, you can run your Web API service and use the Swagger interface to authenticate as previously defined users and test generated endpoints (for example, query available posts). Refer to the following article for additional information [Test the Web API with Swagger or Postman](https://docs.devexpress.com/eXpressAppFramework/404281/backend-web-api-service/test-the-web-api-with-swagger-postman).
+
+![](../../../images/WebAssembly/SwaggerQueryPosts.png)
+
+![](../../../images/WebAssembly/SwaggerPostsResponse.png)
+
+## Create a Blazor WebAssembly application (Frontend)
+
+### Create a new Blazor WebAssembly project
+
+1. Use either the .NET CLI or Visual Studio New Project wizard to add a new Blazor WebAssembly project to the solution.
+
+   ```cmd
+   dotnet new blazorwasm -o Blazor.WebAssembly
+   ```
+
+2. Add the required NuGet packages to the Blazor WebAssembly project.
+
+   ```cmd
+   dotnet add package Blazored.LocalStorage --version 4.3.0
+   dotnet add package Microsoft.AspNetCore.Components.WebAssembly.Authentication --version 7.0.4
+   dotnet add package Microsoft.Extensions.Http --version 7.0.0
+   dotnet add package Microsoft.AspNetCore.Components.WebAssembly.Authentication --version 7.0.4
+   dotnet add package Simple.OData.Client --version 6.0.1
+   dotnet add package DevExpress.Blazor --version 22.2.4
+   ```
+
+### Register Installed Packages Services
```

Comment:
See my comment above regarding the capitalization of headings.

Reply:
Fixed